jemalloc-sys/build.rs

```diff
@@ -28,68 +28,138 @@ fn gnu_target(target: &str) -> String {
-    let src_dir = env::current_dir().unwrap();
-
     if let Some(jemalloc) = env::var_os("JEMALLOC_OVERRIDE") {
+        println!("jemalloc override set");
         let jemalloc = PathBuf::from(jemalloc);
-        println!("cargo:rustc-link-search=native={}",
-                 jemalloc.parent().unwrap().display());
+        println!(
+            "cargo:rustc-link-search=native={}",
+            jemalloc.parent().unwrap().display()
+        );
         let stem = jemalloc.file_stem().unwrap().to_str().unwrap();
         let name = jemalloc.file_name().unwrap().to_str().unwrap();
-        let kind = if name.ends_with(".a") {"static"} else {"dylib"};
+        let kind = if name.ends_with(".a") {
+            "static"
+        } else {
+            "dylib"
+        };
         println!("cargo:rustc-link-lib={}={}", kind, &stem[3..]);
-        return
+        return;
     }
 
     fs::create_dir_all(&build_dir).unwrap();
+    // Disable -Wextra warnings - jemalloc doesn't compile free of warnings with
+    // it enabled: https://github.com/jemalloc/jemalloc/issues/1196
+    let compiler = cc::Build::new().extra_warnings(false).get_compiler();
+    let cflags = compiler
+        .args()
+        .iter()
+        .map(|s| s.to_str().unwrap())
+        .collect::<Vec<_>>()
+        .join(" ");
+    println!("CC={:?}", compiler.path());
+    println!("CFLAGS={:?}", cflags);
+
+    let jemalloc_src_dir = out_dir.join("jemalloc");
+    println!("JEMALLOC_SRC_DIR={:?}", jemalloc_src_dir);
+
+    if jemalloc_src_dir.exists() {
+        std::fs::remove_dir_all(jemalloc_src_dir.clone()).unwrap();
+    }
 
-    let compiler = cc::Build::new().get_compiler();
-    let cflags = compiler.args().iter().map(|s| s.to_str().unwrap())
-                         .collect::<Vec<_>>().join(" ");
+    // Copy jemalloc submodule to the OUT_DIR
+    assert!(out_dir.exists(), "OUT_DIR does not exist");
+    let mut cmd = Command::new("cp");
+    cmd.arg("-rf").arg("jemalloc").arg(jemalloc_src_dir.clone());
+    run(&mut cmd);
 
+    // Configuration files
+    let config_files = ["configure", "VERSION"];
+
+    // Verify that the configuration files are up-to-date
+    if env::var_os("JEMALLOC_SYS_VERIFY_CONFIGURE").is_some() {
+        assert!(!jemalloc_src_dir.join("configure").exists(),
+                "the jemalloc source directory cannot contain configuration files like 'configure' and 'VERSION'");
+        // Run autoconf:
+        let mut cmd = Command::new("autoconf");
+        cmd.current_dir(jemalloc_src_dir.clone());
+        run(&mut cmd);
+
+        for f in &config_files {
+            let mut cmd = Command::new("diff");
+            cmd.arg(jemalloc_src_dir.join(f))
+                .arg(format!("configure/{}", f));
+            run(&mut cmd);
+        }
+    } else {
+        // Copy the configuration files to jemalloc's source directory
+        for f in &config_files {
+            let mut cmd = Command::new("cp");
```

Comment:
I think this could probably safely be done in Rust with `fs::copy`

Reply:
Done.